### PR TITLE
ESYS: Fix deprecated OpenSSL functions.

### PR DESCRIPTION
**Pull request by JuergenReppSIT**
ECC Functions with suffix GFp will become deprecated (DEPRECATED_1_2_0).

Signed-off-by: Juergen Repp <juergen.repp@sit.fraunhofer.de>

### CODE REVIEW

File: src/tss2-esys/esys_crypto_ossl.c [hunk 1]
```diff
@@ -607,6 +622,7 @@ iesys_cryptossl_pk_encrypt(TPM2B_PUBLIC * pub_tpm_key,
     if (1 != RSA_generate_key_ex(rsa_key,
                                  pub_tpm_key->publicArea.parameters.rsaDetail.keyBits,
                                  bne, NULL)) {
+        RAND_set_rand_method(rand_save);
```
Comment (AndreasFuchsSIT) [inline src/tss2-esys/esys_crypto_ossl.c:625]: This is unrelated to the rest of the commit...

**Comment by JuergenReppSIT** [comment]
PR should not be merged also fapi crypto needs a fix.
